packages/redux/src/subscriptions/actions/factories/fetchSubscriptionPackagesFactory.ts

```diff
@@ -0,0 +1,51 @@
+import * as actionTypes from '../../actionTypes';
+import { normalize } from 'normalizr';
+import subscriptionSchema from '../../../entities/schemas/subscriptions';
+import type { Dispatch } from 'redux';
+import type { FetchSubscriptionPackagesAction } from '../../types';
+import type { FetchSubscriptionPackagesFactory } from './types';
+import type {
+  GetSubscriptionPackages,
+  SubscriptionPackage,
+} from '@farfetch/blackout-client/subscriptions/types';
+
+/**
+ * Method to create a thunk factory configured with the specified client.
+ *
+ * @function fetchSubscriptionPackagesFactory
+ * @memberof module:subscriptions/actions/factories
+ *
+ * @param getSubscriptionPackages - Get subscription packages client.
+ *
+ * @returns Thunk factory.
+ */
+const fetchSubscriptionPackagesFactory: FetchSubscriptionPackagesFactory<
+  GetSubscriptionPackages
+> =
+  getSubscriptionPackages =>
+  (query, config) =>
+  async (
+    dispatch: Dispatch<FetchSubscriptionPackagesAction>,
+  ): Promise<SubscriptionPackage> => {
```

Comment:
No need to type the return type as well as it is already set by the `FetchSubscriptionPackagesFactory<GetSubscriptionPackages>` type.